### DISCUSSION TITLE
Eager load $livewire->getOwnerRecord(); relation

### DISCUSSION THREAD

**Discussion by clinvest** [filamentphp/filament#9774] [Help] [answered]
Hi all,

I have a performance issue, it's not slow but it will be a problem.

Is it possible to load a relationship when we load the ownerRecord? Because then I need to access this relation many times (for visibilty reasons). 

For example, in this code, I call two times $program->estimates(). Now in my relation page (which is the edit page of a program (owner record), I have 10 duplicated queries for the same thing. How to handle that?

```
$this->action(function (Form $form): void {
        $this->process(function (array $data, $livewire) {

            $program = $livewire->getOwnerRecord();
            $estimate = $program->estimates()->first();
            $estimate->update($data);

        });

        $this->success();

    });

    $this->visible(static function ($livewire): bool {
        $program = $livewire->getOwnerRecord();
        return $program->estimates()->exists() && !$program->hasApprovedEstimate();
    });
```

Thank you in advance,

**Comment by clinvest** [comment]
I tried to alter the "mount" in the edit ressource but I still have this amount of queries.

**Reply by leandrocfe** [reply, marked as answer]
Did you try to add a property in the relation manager class?

```php

    public $estimates = null;

    public function mount(): void
    {
        parent::mount();
        $this->estimates = $this->getOwnerRecord()->estimates();
    }
```

Then you can use `$livewire->estimates`

**Reply by clinvest** [reply]
Amazing, didn't know how to do that. Thank you very much!

**Comment by clinvest** [comment]
Sorry Leandro, how can I refresh this property after some action?

**Reply by clinvest** [reply]
Sorry, I found.

```
#[On('estimates-updated')]
    public function updateEstimates()
    {
        $this->estimates = $this->program->estimates;
    }
```
